### PR TITLE
FORMS-17206 Reset check box group marks checkbox group field invalidates even if not required

### PR DESCRIPTION
**Pull request by dmaurya929**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js [hunk 1]
```diff
@@ -112,8 +112,13 @@
...
 
         updateValidity(validity) {
-            const valid = validity.valid ? validity.valid : false;
             let widgets = this.widget;
+            if(validity.valid === undefined) {
+                this.element.removeAttribute(FormView.Constants.DATA_ATTRIBUTE_VALID);
+                widgets.forEach(widget => widget.removeAttribute(FormView.Constants.ARIA_INVALID));
+                return;
+            }
+            const valid = validity.valid;
```
Comment (sakshi-arora1) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js:121]: rest of the statements should go in else block

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js [hunk 2]
```diff
@@ -112,8 +112,13 @@
         }
 
         updateValidity(validity) {
-            const valid = validity.valid ? validity.valid : false;
             let widgets = this.widget;
+            if(validity.valid === undefined) {
```
Comment (sakshi-arora1) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js:116]: same would be required for radio button

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js [hunk 3]
```diff
@@ -112,8 +112,13 @@
...
+                this.element.removeAttribute(FormView.Constants.DATA_ATTRIBUTE_VALID);
+                widgets.forEach(widget => widget.removeAttribute(FormView.Constants.ARIA_INVALID));
+                return;
+            }
+            const valid = validity.valid;
             this.element.setAttribute(FormView.Constants.DATA_ATTRIBUTE_VALID, valid);
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js:122]: Can we try to re-use the code ? Let's wrap setting/removing these aria and data attributes in a function and re-use it in both if/else statement

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js (hunk 1 above)
Comment (dmaurya929) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js:121]: yes, that's why used void return statement

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js (hunk 2 above)
Comment (dmaurya929) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js:116]: Okay, let me update.

**Review by sakshi-arora1 — APPROVED** [review]

**Review by rismehta — APPROVED** [review]

**Pull request merged**